{
    "type": "error",
    "data": "Error executing verb \"text_embed\" in create_final_entities: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "stack": "Traceback (most recent call last):\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/datashaper/workflow/workflow.py\", line 415, in _execute_verb\n    result = await result\n             ^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 105, in text_embed\n    return await _text_embed_in_memory(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 130, in _text_embed_in_memory\n    result = await strategy_exec(texts, callbacks, cache, strategy_args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 62, in run\n    embeddings = await _execute(llm, text_batches, ticker, semaphore)\n                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 106, in _execute\n    results = await asyncio.gather(*futures)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 100, in embed\n    chunk_embeddings = await llm(chunk)\n                       ^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/caching_llm.py\", line 96, in __call__\n    result = await self._delegate(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 177, in __call__\n    result, start = await execute_with_retry()\n                    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 159, in execute_with_retry\n    async for attempt in retryer:\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 166, in __anext__\n    do = await self.iter(retry_state=self._retry_state)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 153, in iter\n    result = await action(retry_state)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/_utils.py\", line 99, in inner\n    return call(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 418, in exc_check\n    raise retry_exc.reraise()\n          ^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 185, in reraise\n    raise self.last_attempt.result()\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 449, in result\n    return self.__get_result()\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 401, in __get_result\n    raise self._exception\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 165, in execute_with_retry\n    return await do_attempt(), start\n           ^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 151, in do_attempt\n    await sleep_for(sleep_time)\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 147, in do_attempt\n    return await self._delegate(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 50, in __call__\n    return await self._invoke(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 54, in _invoke\n    output = await self._execute_llm(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/openai/openai_embeddings_llm.py\", line 36, in _execute_llm\n    embedding = await self.client.embeddings.create(\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/resources/embeddings.py\", line 236, in create\n    return await self._post(\n           ^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1831, in post\n    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1525, in request\n    return await self._request(\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1626, in _request\n    raise self._make_status_error_from_response(err.response) from None\nopenai.RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}\n",
    "source": "Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "details": null
}
{
    "type": "error",
    "data": "Error executing verb \"text_embed\" in create_final_entities: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "stack": "Traceback (most recent call last):\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/datashaper/workflow/workflow.py\", line 415, in _execute_verb\n    result = await result\n             ^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 105, in text_embed\n    return await _text_embed_in_memory(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 130, in _text_embed_in_memory\n    result = await strategy_exec(texts, callbacks, cache, strategy_args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 62, in run\n    embeddings = await _execute(llm, text_batches, ticker, semaphore)\n                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 106, in _execute\n    results = await asyncio.gather(*futures)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 100, in embed\n    chunk_embeddings = await llm(chunk)\n                       ^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/caching_llm.py\", line 96, in __call__\n    result = await self._delegate(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 177, in __call__\n    result, start = await execute_with_retry()\n                    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 159, in execute_with_retry\n    async for attempt in retryer:\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 166, in __anext__\n    do = await self.iter(retry_state=self._retry_state)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 153, in iter\n    result = await action(retry_state)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/_utils.py\", line 99, in inner\n    return call(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 418, in exc_check\n    raise retry_exc.reraise()\n          ^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 185, in reraise\n    raise self.last_attempt.result()\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 449, in result\n    return self.__get_result()\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 401, in __get_result\n    raise self._exception\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 165, in execute_with_retry\n    return await do_attempt(), start\n           ^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 151, in do_attempt\n    await sleep_for(sleep_time)\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 147, in do_attempt\n    return await self._delegate(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 50, in __call__\n    return await self._invoke(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 54, in _invoke\n    output = await self._execute_llm(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/openai/openai_embeddings_llm.py\", line 36, in _execute_llm\n    embedding = await self.client.embeddings.create(\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/resources/embeddings.py\", line 236, in create\n    return await self._post(\n           ^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1831, in post\n    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1525, in request\n    return await self._request(\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1626, in _request\n    raise self._make_status_error_from_response(err.response) from None\nopenai.RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}\n",
    "source": "Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "details": null
}
{
    "type": "error",
    "data": "Error executing verb \"text_embed\" in create_final_entities: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "stack": "Traceback (most recent call last):\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/datashaper/workflow/workflow.py\", line 415, in _execute_verb\n    result = await result\n             ^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 105, in text_embed\n    return await _text_embed_in_memory(\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/text_embed.py\", line 130, in _text_embed_in_memory\n    result = await strategy_exec(texts, callbacks, cache, strategy_args)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 62, in run\n    embeddings = await _execute(llm, text_batches, ticker, semaphore)\n                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 106, in _execute\n    results = await asyncio.gather(*futures)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/index/verbs/text/embed/strategies/openai.py\", line 100, in embed\n    chunk_embeddings = await llm(chunk)\n                       ^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/caching_llm.py\", line 96, in __call__\n    result = await self._delegate(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 177, in __call__\n    result, start = await execute_with_retry()\n                    ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 159, in execute_with_retry\n    async for attempt in retryer:\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 166, in __anext__\n    do = await self.iter(retry_state=self._retry_state)\n         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/asyncio/__init__.py\", line 153, in iter\n    result = await action(retry_state)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/_utils.py\", line 99, in inner\n    return call(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 418, in exc_check\n    raise retry_exc.reraise()\n          ^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/tenacity/__init__.py\", line 185, in reraise\n    raise self.last_attempt.result()\n          ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 449, in result\n    return self.__get_result()\n           ^^^^^^^^^^^^^^^^^^^\n  File \"/usr/lib/python3.11/concurrent/futures/_base.py\", line 401, in __get_result\n    raise self._exception\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 165, in execute_with_retry\n    return await do_attempt(), start\n           ^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 151, in do_attempt\n    await sleep_for(sleep_time)\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/rate_limiting_llm.py\", line 147, in do_attempt\n    return await self._delegate(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 50, in __call__\n    return await self._invoke(input, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/base/base_llm.py\", line 54, in _invoke\n    output = await self._execute_llm(input, **kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/graphrag/llm/openai/openai_embeddings_llm.py\", line 36, in _execute_llm\n    embedding = await self.client.embeddings.create(\n                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/resources/embeddings.py\", line 236, in create\n    return await self._post(\n           ^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1831, in post\n    return await self.request(cast_to, opts, stream=stream, stream_cls=stream_cls)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1525, in request\n    return await self._request(\n           ^^^^^^^^^^^^^^^^^^^^\n  File \"/home/k163/codeathon/amr-2024-team-nassar/literature_analysis/.venv/lib/python3.11/site-packages/openai/_base_client.py\", line 1626, in _request\n    raise self._make_status_error_from_response(err.response) from None\nopenai.RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}\n",
    "source": "Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}",
    "details": null
}
